types/ramda/index.d.ts

```diff
@@ -2536,6 +2643,22 @@ export function scan<T, TResult>(fn: (acc: TResult, elem: T) => any, acc: TResul
 export function scan<T, TResult>(fn: (acc: TResult, elem: T) => any, acc: TResult): (list: readonly T[]) => TResult[];
 export function scan<T, TResult>(fn: (acc: TResult, elem: T) => any): (acc: TResult, list: readonly T[]) => TResult[];
 
+// NOTE: this definition isn't particularly useful, but at least it exists
+/**
+ * Transforms a Traversable of Applicative into an Applicative of Traversable.
+ * Dispatches to the sequence method of the second argument, if present.
+ *
+ * See also {@link traverse}.
+ *
+ * @example
+ * ```typescript
+ * R.sequence(R.of, [[1], [2], [3]]); // => [[1, 2, 3]]
+ * R.sequence(R.of)([[1], [2], [3]]); // => [[1, 2, 3]]
+ * ```
+ */
+export function sequence<A>(of: (a: A) => A[], traversable: A[][]): A[][];
```

Comment:
This `sequence`'s type is not precise. Array is just a Applicative Functor and a Traversable structure by chance.

Reply:
Yeah... I don't think it's possible to support Fantasy Land types though. Not sure what to do... maybe type them as `any` like some of the other functions?

Ideally there should be a way to warn the user, but I don't believe there's any way - the only thing I'm aware of that is noticeable would be deprecating it, and this is definitely not deprecated

Reply:
I think maybe there should at least be *something* there, if only just to stop DangerBot from complaining.
Not sure what exactly should be there though.

Reply:
Maybe @essenmitsosse @valerii15298 can give some advice?

Reply:
> I think maybe there should at least be _something_ there, if only just to stop DangerBot from complaining. Not sure what exactly should be there though.

Agree.

Reply:
The question is, what the goal is. If you want to express "We know this function exists, but you should using it in TypeScript" you could go with @deprecated and/or some flavour of using `never`. 

If you want people to be able to use it (right now, TypeScript for example wouldn't auto import the function, when there is no type) but make it clear to the user that he is in danger land, since we recognise the existence of the function, but don't offer any types I don't think that can be expressed in TypeScript in a non-hacky way. You could type it as `unknown` or something, but that's really not how TypeScript is supposed to work I'd say. If you don't type it, it doesn't exist to TypeScript. So leaving the type out is fine. 

As annoying as DangerBot is, I wouldn't want to worsen the user experience of actually using the library, just to slightly increase the developer experience. If anything, we should as the DefinitelyTyped to add some kind of "won't-do-mute-this" switch.

Reply:
I do think people should be able to use it though... just because we cannot type it correctly doesn't mean you should just completely not be able to use this function when using TypeScript.

The real question is, how should we convey to the user that we don't support typings for this function in TypeScript?
I tried using a branded type but it doesn't seem like that would work nicely.
Maybe this would be ok:
```typescript
export function sequence(of: (value: any) => unknown, traversable: unknown): unknown;
```
Not ideal maybe, but it should be (almost) completely safe - the only unsafe part being the `(value: any)`, and since the return type is `unknown` the user should not be able to do anything with it without a type assertion (which would mean that they are intentionally using this function).

Could also add a warning to the JSDoc - if that's *still* not visible enough there are potentially other options

Reply:
I tried to understand practice use of this function but with all those `Just` and `Nothing` cannot...
Basically as I see from example it is just unpacking arrays inside of array.
So if we wanna say that resulting array should be array consisting of 1 array of `A` and travesable is Array of 1 length arrays we can try something like this:
```ts
export function sequence<A>(of: (a: A) => A[], traversable: Array<[A]>): [Array<A>];
```
Did not test it though, hope that works but I am not sure if I understand goal of this function correctly

Reply:
> It's just a coincidence that Array can be regard as an Applicative Functor and a Traversable structure.

As we can't type `sequence` precisely enough in TS now, It might be better to hold off on adding it to DT.